src/repo/github.go

```diff
@@ -0,0 +1,51 @@
+package repo
+
+import (
+	"fmt"
+	"net/http"
+	"bytes"
+	"time"
+	"encoding/json"
+)
+
+type TokenIssuer interface {
+	GeneratedJwtToken() (string, error)
+}
+type GithubClient struct {
+	tokenIssuer TokenIssuer
+}
+func NewGithubClient(tokenIssuer TokenIssuer) *GithubClient {
+	return &GithubClient{
+	  tokenIssuer: tokenIssuer,
+	}
+  }
+
+func (c *GithubClient)  IssueAccessToken(installationID string) (string,error) {
+	jwtToken,err := c.tokenIssuer.GeneratedJwtToken()
+	if(err != nil){
+	return "",fmt.Errorf("%s",err)
+	}
+	url :=fmt.Sprintf("https://api.github.com/app/installations/%s/access_tokens",installationID)
+	req, err := http.NewRequest("POST", url,bytes.NewBuffer(nil))
+	req.Header.Set("Authorization","Bearer"+jwtToken)
+	req.Header.Set("Accept", "application/vnd.github+json")
+	req.Header.Set("X-GitHub-Api-Version", "2022-11-28")
+	
+	client := &http.Client{Timeout: 10 * time.Second}
+	resp, err := client.Do(req)
+	if err != nil {
+			return "", fmt.Errorf("failed to execute request: %w", err)
+	}
+	defer resp.Body.Close()
+	
+	if resp.StatusCode != http.StatusOK {
```

Comment:
check the api spec:
https://docs.github.com/en/rest/apps/apps?apiVersion=2022-11-28#create-an-installation-access-token-for-an-app

it says 201 on success.
so I would check `resp.StatusCode >= 299`.